### PR TITLE
windows demo

### PR DESCRIPTION
**Pull request by unionfindbee**
Adds another independent feature to the demo to show how to demo Mayhem For Code on Windows.

### CODE REVIEW

File: geofencing/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+- **readGPSPointsFromFile**: Reads a file containing GPS points and returns a list of points along with a car identifier.
+- **logGPSPoints**: Logs GPS points and the car identifier.
+- **Main Function**: Processes input files, reads GPS points, logs data, and checks if the car is within the boundary.
+
+## Usage Example
+
+Run the executable with a file containing GPS points and a car identifier, formatted as follows:
+
+```
+40.748817 -73.985428
```
Comment (vlussenburg) [inline geofencing/README.md:69]: Reference the file

File: geofencing/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Windows Geofencing Demo with Mayhem
```
Comment (vlussenburg) [inline geofencing/README.md:1]: Would name the directory different - thinking `cli`. Can we roll the README.md into the top level README? Especially the Mayhem instructions seem redundant.

File: geofencing/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+```
+   mayhem login https://app.mayhem.security/ <your-authentication-token>
+```
+
+## Project Setup
+
+1. **Open Developer PowerShell for Visual Studio 2022**:
+   Run the script in a Developer PowerShell prompt for Visual Studio 2022. This environment includes all necessary environment variables and tools for building the project.
+
+## Build and Run Instructions
```
Comment (vlussenburg) [inline geofencing/README.md:22]: Thinking either put instructions README or ps1, not build (code duplication).

File: geofencing/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+   - `--owner`: Specifies your Mayhem workspace name.
+   - `--project`: Names the project for organizational purposes in Mayhem.
+   - `--target`: An identifier associated with the binary, tracking the specific target being fuzzed.
+
+## Source Code Overview
+
+The `geofencing.cpp` file contains the following key components:
+
+- **GPSPoint Struct**: Defines latitude and longitude for GPS points.
```
Comment (vlussenburg) [inline geofencing/README.md:58]: Would put this inline it the CPP code, not README

File: geofencing/run_mcode.ps1 [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# This copies any test files needed for Mayhem's fuzzing process into the package's 'testsuite' subdirectory.
+# The 'testsuite' files contain inputs and scenarios that Mayhem can use to test for edge cases and vulnerabilities.
+cp testsuite/* package/testsuite
+
+# Step 3: Run the packaged executable with Mayhem For Code
+# Initiates the fuzzing session with Mayhem, using the geofencing package.
+# --owner specifies the user or organization that owns the project in Mayhem.
+# --project identifies this particular fuzzing project by name.
+# --target is a label associated with this binary, used to track the specific target being fuzzed.
+mayhem run package --owner forallsecure-demo --project geofencing --target geofencing
```
Comment (vlussenburg) [inline geofencing/run_mcode.ps1:23]: Demo workspace is `platform-demo`. Match project `mayhem-demo` and target `cli` (or however we'll call it)

Comment (unionfindbee) [reply geofencing/run_mcode.ps1]: Updated!

File: geofencing/run_mcode.ps1 [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Note: This script should be run in a Developer PowerShell for Visual Studio 2022 prompt.
+# This ensures that all necessary environment variables and tools (like MSBuild) are available.
+
+# Step 1: Build the solution using MSBuild
+# This compiles the 'geofencing.sln' solution file, generating an executable.
+msbuild geofencing.sln
```
[the remark this answers is not in the document]
Comment (unionfindbee) [reply geofencing/run_mcode.ps1]: Yes, quite possibly

File: geofencing/README.md (hunk 1 above)
Comment (unionfindbee) [reply geofencing/README.md]: Referenced!

File: geofencing/README.md (hunk 3 above)
Comment (unionfindbee) [reply geofencing/README.md]: Removed duplication from README.

File: geofencing/README.md (hunk 2 above)
Comment (unionfindbee) [reply geofencing/README.md]: Renamed to cli! Rolled into top level README

File: geofencing/run_mcode.ps1 (hunk 6 above)
[the remark this answers is not in the document]
Comment (unionfindbee) [reply geofencing/run_mcode.ps1]: It is possible. Added to Github Actions!

**Review by vlussenburg — APPROVED** [review]

File: cli/testsuite/coordinates.txt
Comment (dbrumley) [inline cli/testsuite/coordinates.txt:1]: This is probably fine.  I did put coordinates in a separate file for the original because then SE and AFL can mutate each one independently.  

(AFL/SE don't recognize line breaks as being separate inputs, so it will try to modify the whole file at once potentially.)

File: .github/workflows/cli.yml
Comment (dbrumley) [inline .github/workflows/cli.yml:1]: nice!  

For internal knowledge, would the existing GPS app also work on windows?

File: README.md [hunk 7]
```diff
@@ -254,6 +254,32 @@ And that’s it! You should be able to see results for Mayhem for Code in your p
   - `--duration 1800` tells Mayhem to run analysis for up to 30 minutes. (If you
     leave this off, Mayhem will continually pentest your app.)
 
+### Step 4D: Run Mayhem for Code to analyze Windows applications
```
Comment (dbrumley) [inline README.md:257]: Please also add a link to the file geofencing.cpp in the first section of the readme, similar to gps_uploader.c and main.py

**Review by dbrumley — APPROVED** [review]
Nice job!  I marked approve to avoid another RTT as my comments are very minor.

File: .github/workflows/cli.yml
Comment (unionfindbee) [reply .github/workflows/cli.yml:1]: Not currently, but with some modifications it could be. 

Some of the libraries and file operations would need to be replaced with Windows equivalents.
To make it a Microsoft Visual Studio demo, we want to add a project & solution file.

File: README.md (hunk 7 above)
Comment (unionfindbee) [reply README.md:261]: Updated!

File: cli/testsuite/coordinates.txt
Comment (unionfindbee) [reply cli/testsuite/coordinates.txt:1]: Oh interesting.  I'll keep it as so for this pull request since Mayhem was able to identify the vulnerabilities. But I will try it out with coordinates in my next update in a separate file to see if it works better.
